pkg/kubelet/server/stats/volume_stat_calculator_test.go

```diff
@@ -141,3 +162,40 @@ func expectedFSStats() kubestats.FsStats {
 		InodesUsed:     &inodesUsed,
 	}
 }
+
+// Fake block-volume/metrics provider, block-devices have no inodes
+var _ volume.BlockVolume = &fakeBlockVolume{}
+
+type fakeBlockVolume struct{}
+
+func (v *fakeBlockVolume) GetGlobalMapPath(*volume.Spec) (string, error) { return "", nil }
+
+func (v *fakeBlockVolume) GetPodDeviceMapPath() (string, string) { return "", "" }
+
+func (v *fakeBlockVolume) GetMetrics() (*volume.Metrics, error) {
+	return expectedBlockMetrics(), nil
+}
+
+func expectedBlockMetrics() *volume.Metrics {
+	return &volume.Metrics{
+		Available: resource.NewQuantity(available, resource.BinarySI),
+		Capacity:  resource.NewQuantity(capacity, resource.BinarySI),
+		Used:      resource.NewQuantity(available-capacity, resource.BinarySI),
+	}
+}
+
+func expectedBlockStats() kubestats.FsStats {
+	metric := expectedBlockMetrics()
+	available := uint64(metric.Available.Value())
+	capacity := uint64(metric.Capacity.Value())
+	used := uint64(metric.Used.Value())
+	null := uint64(0)
+	return kubestats.FsStats{
+		AvailableBytes: &available,
+		CapacityBytes:  &capacity,
+		UsedBytes:      &used,
+		Inodes:         &null,
+		InodesFree:     &null,
+		InodesUsed:     &null,
+	}
+}
```

Comment:
It would be nice to have https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/volume_metrics.go#L204 updated to cover block volumes.

Reply:
Yes, indeed! However it seems that this test is only run for providers "gce", "gke", "aws". I could probably implement basic metrics like capacity (disk-size) for those, but nothing more. Is that what you would like to see?

Reply:
That should be fine.  I am thinking we may need some csi e2es for this too. There are other issues filed about these metrics missing in csi drivers and having these e2es in csi testsuite will ensure that it is caught properly. But `GET_VOLUME_STATS` is optional so it has to be configurable via driver configuration. I think that can happen in a follow up PR/issue.